msticpy/data/drivers/splunk_driver.py

```diff
@@ -214,6 +221,28 @@ def _get_connect_args(
                 title="no Splunk connection parameters",
             )
 
+        if "bearer_token" in self._required_params:
+            epoch_now = datetime.now().timestamp()
+            alg = jwt.get_unverified_header(cs_dict["bearer_token"])["alg"]
+            decoded_token = jwt.decode(
+                cs_dict["bearer_token"],
+                algorithms=[alg],
+                options={"verify_signature": False},
+            )
+            token_idp = decoded_token.get("idp")
+
+            if token_idp == "Splunk":
+                epoch_issued_at = decoded_token["iat"]
+                epoch_expiration = decoded_token["exp"]
+                token_exp = datetime.fromtimestamp(epoch_expiration, timezone.utc)
+                if epoch_issued_at < epoch_now < epoch_expiration:
+                    print(f"This bear_token is active until {token_exp}")
+                else:
+                    raise MsticpyUserConfigError(
+                        f"This bear_token has been expired on {token_exp}",
```

Comment:
I think this should be 'bearer_token'.
I think exceptions.MsticpyConnectionError is probably a better exception to throw, since it isn't really a configuration issue.

Reply:
Thanks for your advice!  I changed it so.